src/fsharp/FSharp.Core/printf.fs

```diff
@@ -1267,17 +1267,51 @@ module internal PrintfImpl =
                 let v = get(key.Value)
                 Cache<'T, 'State, 'Residue, 'Result>.last <- (key.Value, v)
                 v
+    
+    let [<Literal>] MAX_BUILDER_SIZE = 360
+    type StringBuilderCache() =
+        // The value 360 was chosen in discussion with performance experts as a compromise between using
+        // as litle memory (per thread) as possible and still covering a large part of short-lived
+        // StringBuilder creations on the startup path of VS designers.
+       
+        [<ThreadStatic; DefaultValue>]
+        static val mutable private CachedInstance: StringBuilder
+ 
+        static member Acquire(capacity) =
+            if capacity <= MAX_BUILDER_SIZE then
+                let sb = StringBuilderCache.CachedInstance
+                if not (isNull sb) then
+                    // Avoid stringbuilder block fragmentation by getting a new StringBuilder
+                    // when the requested size is larger than the current capacity
+                    if capacity <= sb.Capacity then
+                        StringBuilderCache.CachedInstance <- null
+                        sb.Clear() |> ignore
+                        sb
+                    else
+                        new StringBuilder(capacity)
+                else
+                    new StringBuilder(capacity)
+            else
+                new StringBuilder(capacity)
```

Comment:
```fsharp
    static member Acquire(?capacity: int) =
        match defaultArg capacity DefaultCapacity, StringBuilderCache.CachedInstance with
        | capacity, null -> StringBuilder(capacity)
        | capacity, sb when capacity > MAX_BUILDER_SIZE || capacity > sb.Capacity -> StringBuilder(capacity)
        | _, sb ->
            StringBuilderCache.CachedInstance <- null
            sb.Clear()
```

Reply:
is it different to what I have? or just the same with pattern matching?

Reply:
same